lib/vmselectapi/server.go

```diff
@@ -883,7 +883,9 @@ func (s *Server) processSearch(ctx *vmselectRequestCtx) error {
 				return fmt.Errorf("cannot parse __generation_id label value: %s : %w", generationIDTag.Value, err)
 			}
 			ctx.mb.GenerationID = id
-			ctx.mb.MetricName = mn.Marshal(ctx.mb.MetricName[:0])
+			// pass nil instead of ctx.mb.MetricName[:0] to avoid
+			// iterator MetricBlockRef corruption
+			ctx.mb.MetricName = mn.Marshal(nil)
```

Comment:
This will result in additional memory allocations at this line.
The better solution is to fix the original issue, so ctx.mb.MetricName doesn't point to MetricBlockRef. See the fix at https://github.com/VictoriaMetrics/VictoriaMetrics/commit/916f1ab86c0e5c82295b6eddc7f49401ea4c9a91 .